interpreter/shared/src/test/scala/sigma/serialization/DataSerializerSpecification.scala

```diff
@@ -23,55 +23,74 @@ import scala.reflect.ClassTag
 
 class DataSerializerSpecification extends SerializationSpecification {
 
-  def roundtrip[T <: SType](obj: T#WrappedType, tpe: T) = {
-    val w = SigmaSerializer.startWriter()
-    DataSerializer.serialize(obj, tpe, w)
-    val bytes = w.toBytes
-    val r = SigmaSerializer.startReader(bytes)
-    val res = DataSerializer.deserialize(tpe, r)
-    res shouldBe obj
-
-    val es = CErgoTreeEvaluator.DefaultEvalSettings
-    val accumulator = new CostAccumulator(
-      initialCost = JitCost(0),
-      costLimit = Some(JitCost.fromBlockCost(es.scriptCostLimitInEvaluator)))
-    val evaluator = new CErgoTreeEvaluator(
-      context = null,
-      constants = ErgoTree.EmptyConstants,
-      coster = accumulator, DefaultProfiler, es)
-    val ok = DataValueComparer.equalDataValues(res, obj)(evaluator)
-    ok shouldBe true
-
-    val randomPrefix = arrayGen[Byte].sample.get
-    val r2 = SigmaSerializer.startReader(randomPrefix ++ bytes, randomPrefix.length)
-    val res2 = DataSerializer.deserialize(tpe, r2)
-    res2 shouldBe obj
+  def roundtrip[T <: SType](obj: T#WrappedType, tpe: T, withVersion: Option[Byte] = None) = {
+
+    def test() = {
+      val w = SigmaSerializer.startWriter()
+      DataSerializer.serialize(obj, tpe, w)
+      val bytes = w.toBytes
+      val r = SigmaSerializer.startReader(bytes)
+      val res = DataSerializer.deserialize(tpe, r)
+      res shouldBe obj
+
+      val es = CErgoTreeEvaluator.DefaultEvalSettings
+      val accumulator = new CostAccumulator(
+        initialCost = JitCost(0),
+        costLimit = Some(JitCost.fromBlockCost(es.scriptCostLimitInEvaluator)))
+      val evaluator = new CErgoTreeEvaluator(
+        context = null,
+        constants = ErgoTree.EmptyConstants,
+        coster = accumulator, DefaultProfiler, es)
+      val ok = DataValueComparer.equalDataValues(res, obj)(evaluator)
+      ok shouldBe true
+
+      val randomPrefix = arrayGen[Byte].sample.get
+      val r2 = SigmaSerializer.startReader(randomPrefix ++ bytes, randomPrefix.length)
+      val res2 = DataSerializer.deserialize(tpe, r2)
+      res2 shouldBe obj
+    }
+
+    withVersion match {
+      case Some(ver) =>
+        VersionContext.withVersions(ver, 1) {
```

Comment:
Should be `0` since script versions are zero based.
Also  better pass by named argument, as the meaning is not clear without IDE hints.